zipline/finance/performance/position_tracker.py

```diff
@@ -203,20 +290,36 @@ def pay_dividends(self, dividend_frame):
             stock = row['payment_sid']
             share_count = row['share_count']
             # note we create a Position for stock dividend if we don't
-            # already own the security
+            # already own the asset
             position = self.positions[stock]
 
             position.amount += share_count
             self._position_amounts[stock] = position.amount
             self._position_last_sale_prices[stock] = position.last_sale_price
+            self._update_asset(stock)
+            self._invalidate_cache()
 
         # Add cash equal to the net cash payed from all dividends.  Note that
-        # "negative cash" is effectively paid if we're short a security,
+        # "negative cash" is effectively paid if we're short an asset,
         # representing the fact that we're required to reimburse the owner of
         # the stock for any dividends paid while borrowing.
         net_cash_payment = payments['cash_amount'].fillna(0).sum()
         return net_cash_payment
 
+    def _generate_end_sid_transaction(self, sid, dt):
```

Comment:
Where is this called from?

Can we be doing this check elsewhere rather than inside the method (e.g. `if tracker.has_position(sid): tracker._generate_end_sid_transaction)`)?  

If not, can we rename this to something like `_maybe_create_end_sid_transaction` so it's clearer that it might return `None`